uhttpd: add audio/video support for apple airplay

komenski opened the pull request:
Airplay of a statically hosted video file from one Apple device to another fails due to unrecognized content-type. Adding Airplay supported mime types fixes the issue.

Signed-off-by: Sasha Andonov <s.andonnov@gmail.com>

neheb commented:
make package/uhttpd/{clean,refresh}

Needs a PKG_RELEASE bump too.

neheb commented:
maybe @Ansuel can commit this directly.

Ansuel commented:
@komenski i didn't understand what is the problem from the commit description. Can you provide more info?

komenski commented:
> @komenski i didn't understand what is the problem from the commit description. Can you provide more info?

Short story long: let's say we're hosting a video file at _url1_: http://openwrt.local/luci-static/video.mp4. If url1 is passed to a HTML5 browser, it will download the file instead of playing it back as it thinks it is a binary (as we dont have an entry for mp4 extension here). Now, that's easily worked around by wrapping _url1_ in HTML5 <video> tag, inside of an html file which we put at _url2_ http://openwrt.local/luci-static/video.html, and the browser starts the playback as it now knows it is a video. However, if we now wanted to send the video over Airplay to a second device, it's not gonna send the html file found at url2, but rather the video file found at url1, and the playback on second device will fail as it thinks it is a binary. How do I know that? I don't know what Airplay does under the hood, but I know I'm getting the same result if I remove mp4 mime type from lighttpd config - hence, this must be the fix.

Ansuel commented:
Ok nice explaination. Would be good to add this to the commit description

komenski commented:
> Ok nice explaination. Would be good to add this to the commit description

Let me see if I can rephrase

komenski committed:
uhttpd: add audio/video support for apple airplay

Airplay of a statically hosted video file from one Apple device to
another fails due to unrecognized content-type.

Let's assume we have url1: http://openwrt.local/luci-static/video.mp4.
If url1 is passed to a HTML5 browser, it will download the file instead
of playing it back as it thinks it is a binary (no entry for mp4
extension atm). Let's also assume we worked this around by wrapping url1
in HTML5 video tag, inside of an html file which we will put at
url2: http://openwrt.local/luci-static/video.html. The playback starts as
the browser now knows it is a video. However, if we now wanted to send
the video over Airplay to a second device, it's not gonna share the html
file found at url2, but rather the video file found at url1, and the
playback on second device will fail as it thinks it is a binary.

Adding Airplay supported mime types fixes the issue.

Signed-off-by: Sasha Andonov <s.andonnov@gmail.com>

hauke commented:
Thank you for the patch. I applied it directly to the uhttpd repository, see https://git.openwrt.org/?p=project/uhttpd.git;a=commitdiff;h=47561aa13574068403d48f13ea310f8511057b2b and then updated uhttpd to this version in master, see d14559e9df4052cfaecd016c2afd2353ce18c455
